src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/instance_trial_of_the_crusader.cpp

```diff
@@ -59,6 +59,42 @@ class instance_trial_of_the_crusader : public InstanceMapScript
         uint64 GO_WebDoorGUID;
         uint64 GO_FloorGUID;
 
+        void SpawnAnubArak()
+		{
+			if (Creature* barrett = instance->GetCreature(NPC_BarrettGUID))
+			{
+				barrett->SetVisible(false);
+				if (!barrett->FindNearestCreature(NPC_ANUBARAK, 99999.0f))
+					barrett->SummonCreature(NPC_ANUBARAK, Locs[LOC_ANUB].GetPositionX(), Locs[LOC_ANUB].GetPositionY(), Locs[LOC_ANUB].GetPositionZ(), Locs[LOC_ANUB].GetOrientation(), TEMPSUMMON_CORPSE_TIMED_DESPAWN, 630000000);
+			}
+
+			// move corpses:
+			if (Creature* c = instance->GetCreature(NPC_IcehowlGUID))
+			{
+				c->UpdatePosition(626.57f, 162.8f, 140.25f, 4.44f, true);
+				c->StopMovingOnCurrentPos();
+				c->DestroyForNearbyPlayers();
+			}
+			if (Creature* c = instance->GetCreature(NPC_JaraxxusGUID))
+			{
+				c->UpdatePosition(603.92f, 102.61f, 141.85f, 1.4f, true);
+				c->StopMovingOnCurrentPos();
+				c->DestroyForNearbyPlayers();
+			}
+			if (Creature* c = instance->GetCreature(NPC_LightbaneGUID))
+			{
+				c->UpdatePosition(634.58f, 147.16f, 140.5f, 3.02f, true);
+				c->StopMovingOnCurrentPos();
+				c->DestroyForNearbyPlayers();
+			}
+			if (Creature* c = instance->GetCreature(NPC_DarkbaneGUID))
+			{
+				c->UpdatePosition(630.88f, 131.39f, 140.8f, 3.02f, true);
+				c->StopMovingOnCurrentPos();
+				c->DestroyForNearbyPlayers();
+			}
```

Comment:
what about using a switch-case instead of multiple-ifs?

also those lines are repeated:

```
c->StopMovingOnCurrentPos();
c->DestroyForNearbyPlayers();
```

Reply:
```
switch (c->GetGUID())
{
    case NPC_IcehowlGUID:
        c->UpdatePosition(626.57f, 162.8f, 140.25f, 4.44f, true);
        c->StopMovingOnCurrentPos();
        c->DestroyForNearbyPlayers();
        break;
    case NPC_JaraxxusGUID:
        c->UpdatePosition(603.92f, 102.61f, 141.85f, 1.4f, true);
        c->StopMovingOnCurrentPos();
        c->DestroyForNearbyPlayers();
        break;
    case NPC_LightbaneGUID:
        c->UpdatePosition(634.58f, 147.16f, 140.5f, 3.02f, true);
        c->StopMovingOnCurrentPos();
        c->DestroyForNearbyPlayers();
        break;
    case NPC_DarkbaneGUID:
        c->UpdatePosition(630.88f, 131.39f, 140.8f, 3.02f, true);
        c->StopMovingOnCurrentPos();
        c->DestroyForNearbyPlayers();
        break;
}
```